src/ng/q.js

```diff
@@ -276,7 +315,7 @@ function qFactory(nextTick, exceptionHandler) {
 
       this.$$state.pending = this.$$state.pending || [];
       this.$$state.pending.push([result, onFulfilled, onRejected, progressBack]);
-      if (this.$$state.status > 0) scheduleProcessQueue(this.$$state);
+      if (this.$$state.status > 0) scheduleProcessQueue(result.promise, this.$$state);
```

Comment:
I understand what you mean, and I appreciate the effort you put into your comment reply, but the line you're pointing to doesn't change any of that. The only reason the promise is passed into the function is so that a particular property can be read later on by it.

I still might be missing something. I think the only way this is going to be clear is to have a test that proves that `case 2` fails.